lib/SILOptimizer/Utils/ConstExpr.cpp

```diff
@@ -539,6 +557,39 @@ ConstExprFunctionState::computeCallResult(ApplyInst *apply) {
 
   SILFunction *callee = calleeFn.getFunctionValue();
 
+  // If this is a well-known function, do not step into it.
+  switch (classifyFunction(callee)) {
+  default:
+    break;
+  case WellKnownFunction::StringInitEmpty: { // String.init()
+    assert(conventions.getNumDirectSILResults() == 1 &&
+           conventions.getNumIndirectSILResults() == 0 &&
+           "unexpected String.init() signature");
+    auto result = SymbolicValue::getString("", evaluator.getASTContext());
+    setValue(apply, result);
+    return None;
+  }
+  case WellKnownFunction::StringMakeUTF8: {
+    // String.init(_builtinStringLiteral start: Builtin.RawPointer,
+    //             utf8CodeUnitCount: Builtin.Word,
+    //             isASCII: Builtin.Int1)
+    assert(conventions.getNumDirectSILResults() == 1 &&
+           conventions.getNumIndirectSILResults() == 0 &&
+           conventions.getNumParameters() == 4 && "unexpected signature");
+    auto literal = getConstantValue(apply->getOperand(1));
+    if (literal.getKind() != SymbolicValue::String)
+      break;
+    auto literalVal = literal.getStringValue();
+
+    auto byteCount = getConstantValue(apply->getOperand(2));
+    if (byteCount.getKind() != SymbolicValue::Integer ||
+        byteCount.getIntegerValue().getLimitedValue() != literalVal.size())
+      break;
+    setValue(apply, literal);
+    return None;
+  }
+  }
+
```

Comment:
@marcrasi I wonder if it would be better to make a separate function for lines 560 to 591 that handles well-known functions. As we plan to add more well-known functions, it would make this more modular.

Reply:
done!